### PR TITLE
[NTOSKRNL] Use MI_IS_WRITE_ACCESS(), instead of magic values

### PR DESCRIPTION
**Pull request by SergeGautherie**
Explicit macros vs magic values.

### CODE REVIEW

**Review by Extravert-ir — CHANGES_REQUESTED** [review]
1. I don't think this variable is needed at all
2. You're not consistent: not all usages are fixed

**Comment by SergeGautherie** [comment]
>     1. I don't think this variable is needed at all

Removed.

>     2. You're not consistent: not all usages are fixed

Which other usages?

**Comment by Extravert-ir** [comment]
`ntoskrnl/kdbg/kdb.c` for example

**Comment by SergeGautherie** [comment]
> `ntoskrnl/kdbg/kdb.c`

Thanks for the hint!
Done.

**Review by tkreuzer — APPROVED** [review]

File: ntoskrnl/include/internal/arm/mm.h [hunk 1]
```diff
@@ -91,9 +91,16 @@
...
+#define MI_IS_NOT_PRESENT_FAULT(FaultCode)  TRUE
+#define MI_IS_WRITE_ACCESS(FaultCode)       TRUE
+// 0x00000004: user-mode access.
+// TODO: Confirm which value to use.
+#define MI_IS_RESERVED_BIT_FAULT(FaultCode) FALSE
+#define MI_IS_INSTRUCTION_FETCH(FaultCode)  FALSE
+// 0x00000020: protection-key violation.
+// Bits 6-14: reserved.
+// 0x00008000: violation of SGX-specific access-control requirements.
+// Bits 16-31: reserved.
```
Comment (ThFabba) [inline ntoskrnl/include/internal/arm/mm.h:103]: These bits are x86-specific, so this is nonsensical.

Comment (SergeGautherie) [reply ntoskrnl/include/internal/arm/mm.h]: Right.
Replaced with a better TODO.

**Pull request merged**